src/parsers/QASM3Parser.cpp

```diff
@@ -768,12 +775,26 @@ class OpenQasm3Parser final : public InstVisitor {
             ifStatement->debugInfo);
     }
 
-    // translate statements in then block
+    // translate statements in then/else blocks
+    if (!ifStatement->thenStatements.empty()) {
+      auto thenOps = translateBlockOperations(ifStatement->thenStatements);
+      qc->emplace_back(std::make_unique<qc::ClassicControlledOperation>(
+          thenOps, creg->second, rhs->getUInt(), comparisonKind));
+    }
+    if (!ifStatement->elseStatements.empty()) {
+      auto elseOps = translateBlockOperations(ifStatement->elseStatements);
+      qc->emplace_back(std::make_unique<qc::ClassicControlledOperation>(
+          elseOps, creg->second, rhs->getUInt(), flippedComparisonKind));
+    }
+  }
```

Comment:
I believe this is the right way to go for now and a quick win. 👍🏼 
In the long run, it might be worth to just revisit the design of the `ClassicControlledOperation` class; especially given that we plan to support further control flow logic soon.
At that point, we might just relabel/rename the class to something like `IfElseOperation`.
And, then, we can also simply print it as an `if ... else`.